sw/device/silicon_creator/lib/drivers/rnd.c

```diff
@@ -1,30 +1,113 @@
 // Copyright lowRISC contributors.
 // Licensed under the Apache License, Version 2.0, see LICENSE for details.
 // SPDX-License-Identifier: Apache-2.0
-
 #include "sw/device/silicon_creator/lib/drivers/rnd.h"
 
 #include "sw/device/lib/base/abs_mmio.h"
 #include "sw/device/lib/base/csr.h"
 #include "sw/device/lib/base/hardened.h"
+#include "sw/device/lib/base/macros.h"
+#include "sw/device/silicon_creator/lib/crc32.h"
 #include "sw/device/silicon_creator/lib/drivers/otp.h"
 
+#include "entropy_src_regs.h"
 #include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
 #include "otp_ctrl_regs.h"
 #include "rv_core_ibex_regs.h"
 
 enum {
-  kBase = TOP_EARLGREY_RV_CORE_IBEX_CFG_BASE_ADDR,
+  kBaseEntropySrc = TOP_EARLGREY_ENTROPY_SRC_BASE_ADDR,
+  kBaseIbex = TOP_EARLGREY_RV_CORE_IBEX_CFG_BASE_ADDR,
+
+  // This covers the health threshold registers which are contiguous. The alert
+  // threshold register is not included here.
+  kNumHealthRegisters = 9,
 };
 
+// Check the number of health registers covered by this driver.
+static_assert(kNumHealthRegisters ==
+                  (ENTROPY_SRC_EXTHT_LO_THRESHOLDS_REG_OFFSET -
+                   ENTROPY_SRC_REPCNT_THRESHOLDS_REG_OFFSET) /
+                          sizeof(uint32_t) +
+                      1,
+              "foobar");
```

Comment:
nit: can you update the `static_assert` message?

Reply:
Done